dev/reference/best-practices/grafana-monitor.md

```diff
@@ -0,0 +1,206 @@
+
+![Check metric information in history](/media/best-practices/manually-input-check-metric.jpeg)
+
+### Tip 6: Use the `Avg` function
+
+Generally, only `Max` and `Current` functions are available in the legend by default. When the metric information fluctuates greatly, you can add other summary functions such as the `Avg` function to the legend to check the overall trend for the duration of time.
+
+Add summary functions such as the `Avg` function:
+
+![Add summary functions such as Avg](/media/best-practices/add-avg-function.jpeg)
+
+Then check the overall trend:
+
+![Add Avg function to check the overall trend](/media/best-practices/add-avg-function-check-trend.jpg)
+
+### Tip 7: Use the API of Prometheus to obtain the result of query expressions
+
+Grafana obtains data through the API of Prometheus and you can use the API to obtain information as well. Besides, it also has the following usages:
+
+- Automatically obtains information such as cluster size and status.
+- Makes minor changes to the expression to provide information for the report, such as counting the total amount of QPS per day, the peak value of QPS per day, and the response time per day.
+- Performs regular health inspect on the important metrics.
+
+The API of Prometheus is shown as follows:
+
+![The API of Prometheus](/media/best-practices/prometheus-api-interface.jpg)
+
+{{< copyable "shell-regular" >}}
+
+```bash
+curl -u user:pass 'http://__grafana_ip__:3000/api/datasources/proxy/1/api/v1/query_range?query=sum(tikv_engine_size_bytes%7Binstancexxxxxxxxx20181%22%7D)%20by%20(instance)&start=1565879269&end=1565882869&step=30' |python -m json.tool
+```
+
+```
+{
+    "data": {
+        "result": [
+            {
+                "metric": {
+                    "instance": "xxxxxxxxxx:20181"
+                },
+                "values": [
+                    [
+                        1565879269,
+                        "1006046235280"
+                    ],
+                    [
+                        1565879299,
+                        "1006057877794"
+                    ],
+                    [
+                        1565879329,
+                        "1006021550039"
+                    ],
+                    [
+                        1565879359,
+                        "1006021550039"
+                    ],
+                    [
+                        1565882869,
+                        "1006132630123"
+                    ]
+                ]
+            }
+        ],
+        "resultType": "matrix"
+    },
+    "status": "success"
+}
+```
+
+## Summary
+
+The Grafana + Prometheus monitoring platform is a very powerful tool. Making good use of it can improve efficiency, saving you a lot of time on analyzing the status of the TiDB cluster. More importantly, it can help you diagnose problems. This tool is very useful in the operation and maintenance of the TiDB cluster, especially when there is a large amount of data.
```

Comment:
```suggestion
The Grafana + Prometheus monitoring platform is a very powerful tool. Making good use of it can improve efficiency, saving you a lot of time on analyzing the status of the TiDB cluster. More importantly, it can help you diagnose problems. This tool is very useful in the operation and maintenance of TiDB clusters, especially when there is a large amount of data.
```